### PR TITLE
Parsing Log File And Filtering Data

### PR DESCRIPTION
**Pull request by daniyalumer**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"log"
+	"os"
+
+	"github.com/daniyalumer/repeated-ip-report/parser"
+)
+
+type RedirectLog = parser.RedirectLog
```
Comment (mzeeshanali) [inline main.go]: why?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+func main() {
+	file, err := os.Open("logs_1feb.txt")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	uniqueRedirectLogs := make(map[string][]RedirectLog)
+	returnLogs := make(map[string][]RedirectLog)
+
+	parser.ParseFile(file, uniqueRedirectLogs, returnLogs)
```
Comment (mzeeshanali) [inline main.go]: error handling is missing?

File: parser/parser.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+func ParseLine(line string) RedirectLog {
+	timeRegex := regexp.MustCompile(`(\d{2}/\w{3}/\d{4}:\d{2}:\d{2}:\d{2} \+\d{4})`)
+	timeStampStr := timeRegex.FindString(line)
+
+	timeStamp, err := time.Parse("02/Jan/2006:15:04:05 +0000", timeStampStr)
+	if err != nil {
+		log.Fatal(err)
+	}
+
```
Comment (mzeeshanali) [inline parser/parser.go]: explore time.RFC formats in go

File: parser/parser.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	keywordRegex := regexp.MustCompile(`keyword=([^ ]+)`)
+	keyword := keywordRegex.FindString(line)
+
+	userRegex := regexp.MustCompile(`"([^"]+)"$`)
+	userAgent := userRegex.FindString(line)
+
+	urlRegex := regexp.MustCompile(`\s+(\S+)\s+HTTP/1.1`)
+	urlString := urlRegex.FindString(line)
+
```
Comment (mzeeshanali) [inline parser/parser.go]: define regexes at one place and use them where needed

**Review by mzeeshanali — CHANGES_REQUESTED** [review]

File: parser/parser.go (hunk 3 above)
Comment (daniyalumer) [reply parser/parser.go]: I have had a look at them

**Pull request merged**